### PR TITLE
React Todo App : Emotion UI added

### PR DESCRIPTION
**Pull request by shadman0711**
Fixes #

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [x] Improvement
- [ ] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?
Emotion UI elements added

## What was done
* Emotion styled components added

## How to test

- [ ] Needs regression testing (add QA needed label)

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

## Reminder list

- Determine whether REST API Documentation should be updated
- Make an assessment on whether unit tests should be updated and/or added
- Ensure follow up tasks are linked to this PR

QA - if regression testing was requested:

- [ ] QA has verified

### CODE REVIEW

File: todo-app/src/App.js [hunk 1]
```diff
@@ -1,31 +1,17 @@
-import './App.css';
-import { useState } from 'react';
+import { Container } from './components/Container.styled';
+import { ContentArea } from './components/ContentArea.styled';
```
Comment (tusher-A) [inline todo-app/src/App.js:2]: - These two files should not be placed inside `src/components` folder. It doesn't make any sense as you are using it inside `App.js` file.
- Create a file called `App.styled.js` as per your preferred way. Move the `Container` and `ContentArea` there.

File: todo-app/src/components/Container.styled.js [hunk 2]
```diff
@@ -1,11 +1,8 @@
-.Container {
+import styled from '@emotion/styled';
```
Comment (tusher-A) [inline todo-app/src/components/Container.styled.js:1]: Remove this from here. Check comment above.

File: todo-app/src/components/ContentArea.styled.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import styled from '@emotion/styled';
```
Comment (tusher-A) [inline todo-app/src/components/ContentArea.styled.js:1]: Remove this from here. Check comment above.

File: todo-app/src/components/TodoForm/TodoForm.js [hunk 4]
```diff
@@ -1,14 +1,28 @@
-import './TodoForm.css';
-import React, { useState, useRef } from 'react';
+import React, { useState, useRef, useContext } from 'react'
+import { TodoContext } from '../../context/todoContext';
+import { FormContainer } from './FormContainer.styled';
+import { FormInput } from './FormInput.styled';
```
Comment (tusher-A) [inline todo-app/src/components/TodoForm/TodoForm.js:4]: - I don't understand, why are you creating two separate files to incorporate style into the same component. Does that even make sense to you?
-  Both `FormContainer` or `FormInput` can be place to a single file and you can name it `TodoFrom.styled.js` as per your preferred way. 

Please communicate with your fellow trainee if you don't understand this.

File: todo-app/src/components/TodoForm/FormContainer.styled.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import styled from '@emotion/styled';
```
Comment (tusher-A) [inline todo-app/src/components/TodoForm/FormContainer.styled.js:1]: Why single style on single file? I am really confused.

File: todo-app/src/components/TodoForm/FormInput.styled.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import styled from '@emotion/styled';
```
Comment (tusher-A) [inline todo-app/src/components/TodoForm/FormInput.styled.js:1]: Same here again.

File: todo-app/src/components/Todos/TodoList.styled.js [hunk 7]
```diff
@@ -0,0 +1,6 @@
+import styled from '@emotion/styled';
```
Comment (tusher-A) [inline todo-app/src/components/Todos/TodoList.styled.js:1]: Rename the file  to `Todo.styled.js`. Makes more sense.

File: todo-app/src/components/Todos/components/ListItem/ListItem.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+// import { DeleteButton } from '../../../../components/EmotionUIElements'
+import { ListItem as TodoItem, Span } from './ListItem.styled'
+import { ToggleButton } from './ToggleButton.styled';
+import { DeleteButton } from './DeleteButton.styled';
```
Comment (tusher-A) [inline todo-app/src/components/Todos/components/ListItem/ListItem.js:5]: - Again why you have separate styles in separate files, when you can keep them just in one style file. Please fix this. This is really bad practice.

File: todo-app/src/context/todoContext.js [hunk 9]
```diff
@@ -0,0 +1,13 @@
+import { createContext, useState } from 'react';
+
+export const TodoContext = createContext();
+
+export const TodoProvider = ({ children }) => {
+    const [todos, setTodos] = useState([]);
```
Comment (tusher-A) [inline todo-app/src/context/todoContext.js:6]: You were asked to use `useReducer` not `useState`. Please redo this.

**Review by tusher-A — CHANGES_REQUESTED** [review]
Some serious issue remains in the code. Please address them.

File: todo-app/src/context/todoContext.js (hunk 9 above)
Comment (shadman0711) [reply todo-app/src/context/todoContext.js]: Added useReducer in commit [39bd531](https://github.com/CraftsmenLtd/coe-batch-q322/pull/247/commits/39bd53138800493ff01d43f60c36c57addb8077c)

File: todo-app/src/App.js (hunk 1 above)
Comment (shadman0711) [reply todo-app/src/App.js]: Updated file structure in commit [7717cf6](https://github.com/CraftsmenLtd/coe-batch-q322/pull/247/commits/7717cf6fc05ae0b6266281efbe44e54955b077dc)

File: todo-app/src/components/Container.styled.js (hunk 2 above)
Comment (shadman0711) [reply todo-app/src/components/Container.styled.js]: Updated file structure in commit [7717cf6](https://github.com/CraftsmenLtd/coe-batch-q322/pull/247/commits/7717cf6fc05ae0b6266281efbe44e54955b077dc)

File: todo-app/src/components/ContentArea.styled.js (hunk 3 above)
Comment (shadman0711) [reply todo-app/src/components/ContentArea.styled.js]: Removed in commit [7717cf6](https://github.com/CraftsmenLtd/coe-batch-q322/pull/247/commits/7717cf6fc05ae0b6266281efbe44e54955b077dc)

File: todo-app/src/components/TodoForm/TodoForm.js (hunk 4 above)
Comment (shadman0711) [reply todo-app/src/components/TodoForm/TodoForm.js]: Updated file structure in commit [7717cf6](https://github.com/CraftsmenLtd/coe-batch-q322/pull/247/commits/7717cf6fc05ae0b6266281efbe44e54955b077dc)

File: todo-app/src/components/TodoForm/FormContainer.styled.js (hunk 5 above)
Comment (shadman0711) [reply todo-app/src/components/TodoForm/FormContainer.styled.js]: Changed this in commit [7717cf6](https://github.com/CraftsmenLtd/coe-batch-q322/pull/247/commits/7717cf6fc05ae0b6266281efbe44e54955b077dc)

File: todo-app/src/components/TodoForm/FormInput.styled.js (hunk 6 above)
Comment (shadman0711) [reply todo-app/src/components/TodoForm/FormInput.styled.js]: Updated in commit [7717cf6](https://github.com/CraftsmenLtd/coe-batch-q322/pull/247/commits/7717cf6fc05ae0b6266281efbe44e54955b077dc)

File: todo-app/src/components/Todos/TodoList.styled.js (hunk 7 above)
Comment (shadman0711) [reply todo-app/src/components/Todos/TodoList.styled.js]: Renamed in commit [7717cf6](https://github.com/CraftsmenLtd/coe-batch-q322/pull/247/commits/7717cf6fc05ae0b6266281efbe44e54955b077dc)

File: todo-app/src/components/Todos/components/ListItem/ListItem.js (hunk 8 above)
Comment (shadman0711) [reply todo-app/src/components/Todos/components/ListItem/ListItem.js]: Fixed in commit [7717cf6](https://github.com/CraftsmenLtd/coe-batch-q322/pull/247/commits/7717cf6fc05ae0b6266281efbe44e54955b077dc)

**Review by tusher-A — APPROVED** [review]
LGTM.

**Pull request merged**